src/AbstractArray.php

```diff
@@ -420,6 +420,19 @@ public function containsKey($key)
     }
 
     /**
+     * Find the given value in the array using a closure
+     *
+     * @param \Closure $closure
+     * @param mixed $needle
+     *
+     * @return bool Returns true if the given value is found, false otherwise
+     */
+    public function exists(\Closure $closure, $needle)
+    {
+        return $closure($needle, $this->elements);
```

Comment:
We need to use more complex logic. Replace it with https://github.com/doctrine/collections/blob/master/lib/Doctrine/Common/Collections/ArrayCollection.php#L200-L209